src/components/cards/WarningCard.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react'
+import { Platform, View } from 'react-native'
+import IonIcon from 'react-native-vector-icons/Ionicons'
+
+import { fixSides, mapSides, sidesToMargin, sidesToPadding } from '../../util/sides'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+
+type Props = {
+  title: string
+  header?: string
+  points?: string[]
+  footer?: string
+  marginRem?: number[] | number
+  paddingRem?: number[] | number
+}
+/**
+ * A warning card that accepts a title, header, bullet points and a footer,
+ * rendering the bullet points with the correct vertical alignment when
+ * the messages overflow.
+ *  .___________________________.
+ *  | ⚠ Warning                 |
+ *  |                           |
+ *  | This is the header text   |
+ *  |                           |
+ *  | • This is an overflowing  |
+ *  |   bulletpoint message     |
+ *  | • This one's short        |
+ *  |                           |
+ *  | This is the footer text   |
+ *  |___________________________|
+ */
+export function WarningCard({ title, header, points, footer, marginRem, paddingRem }: Props) {
+  const theme = useTheme()
+  const margin = sidesToMargin(mapSides(fixSides(marginRem, 1), theme.rem))
+  const padding = sidesToPadding(mapSides(fixSides(paddingRem, 1), theme.rem))
+
+  const styles = getStyles(theme)
+
+  const renderBulletpoint = (message: string) => {
+    return (
+      <View style={styles.bulletpointRow}>
+        <EdgeText style={styles.bulletpointText}>{'\u2022 '}</EdgeText>
+        <EdgeText style={styles.bulletpointText} numberOfLines={0}>
+          {message}
+        </EdgeText>
+      </View>
+    )
+  }
+  return (
+    <View style={[styles.warning, margin, padding]}>
+      <View style={styles.titleContainer}>
+        <IonIcon
+          name={Platform.OS === 'ios' ? 'ios-warning-outline' : 'md-warning-outline'}
+          style={styles.icon}
+          color={theme.warningText}
+          size={theme.rem(0.8)}
+        />
+        <EdgeText style={styles.title}>{title}</EdgeText>
+      </View>
+      {header && (
+        <EdgeText style={styles.text} numberOfLines={0}>
+          {header}
+        </EdgeText>
+      )}
+      {points && <View style={styles.bulletpoints}>{points?.map(point => renderBulletpoint(point))}</View>}
```

Comment:
Also suggest updating this such that both `undefined` and `[]` `points` result in a hidden bulletpoints view here.